main/res/layout/date_range_selector_view.xml

```diff
@@ -0,0 +1,53 @@
+<?xml version="1.0" encoding="utf-8"?>
```

Comment:
As I guess that you have already a couple of layout files for the new filtering system, and that there will be many more: May it be useful to put them all into a subfolder of `layouts` to reduce that endless list a bit and to group those filter-related files?

Reply:
We also recently discussed styles/themes in our dev-meeting. AFAIR @eddiemuc started with a new base style for the filtering and the idea could be anyway to establish a "second layout setup" (with proper parenting, inheritation, etc.) in parallel and whenever we touch an area we try to bring it onto this cleaner style hierarchy. 

Or is this a whole other topic and I misunderstood your comment?

Reply:
Sadly I've missed that discussion completely due to my not being back in time...

So far I was only speaking of organization of layout files in the file system. Currently we have more than 100 layout files in a linear list, which is pretty annoying to handle when working with layouts, so some grouping may help here, especially if there will be lots of new files.

With regards to new base layout: I've also started with a new base layout, including some nice method for switching between dark and light mode, but was blocked by the fact, that our preferences are based on a different type of activities, not being compatible to `AppCompatActivity`, which would require us to maintain two sets of themes/styles. That's why I opened #10579.

Reply:
@moving-bits 
> As I guess that you have already a couple of layout files for the new filtering system, and that there will be many more: May it be useful to put them all into a subfolder of `layouts` to reduce that endless list a bit and to group those filter-related files?

Correct, we should think about better structuring our layout files. But specifically this one (d`ate_range_selector_view.xml`) is **not** supposed to be cache-filter-specific. In an attempt to make gui components more reusable I created a custom view `DateRangeSelector `which may be used now anywhere one needs to make the user a date-range (currently of course this is only needed in cache filter). 

Up to now I try to use a name scheme for layout files, with the prefix specifying the place of usage (e.g. `cache_filter_`) and the postfix trying to specify the kind of usage (e.g. `_view`, `_activity`, `_fragment`). Consequently you will find all really cache_filter-specific layout files by their prefix.

I suggest to move the topic of reorganizing layout files to its own issue, it goes beyond the scope of cache filters. Would that be ok?

Reply:
> We also recently discussed styles/themes in our dev-meeting. AFAIR @eddiemuc started with a new base style for the filtering and the idea could be anyway to establish a "second layout setup" (with proper parenting, inheritation, etc.) in parallel and whenever we touch an area we try to bring it onto this cleaner style hierarchy.
> 
> Or is this a whole other topic and I misunderstood your comment?

@Lineflyer , @moving-bits 
This discussion was about incorporation https://material.io/ as the base of c:geo styles/themes to apply the design capabilities of this for a GUI rework. But I haven't done amything real on this up to now anyway, only thing I did so far is creating the new themes dark_material and light_material as copies of dark/light but inheriting from the material-themes instead of the appcompat themes. And the new activity CacheFilterActivity is the only one using these new themes so far. 

I can easily undo this again if wanted?

The only views really needing to be based on material thems currently are the new ContinuousRangeSlider and ItemRangeSlider; and those can also directly inherit from material themes (I use ContextThemeWrapper inside anyway). 

So should I remove this stuff for now? (but please in a separate issue)

Reply:
> I suggest to move the topic of reorganizing layout files to its own issue, it goes beyond the scope of cache filters. Would that be ok?

Sure. I just got triggered by seeing your hard work on the new filters and the many layout-related things you do. I'll open a separate issue for organizing the layout files.

Reply:
> In an attempt to make gui components more reusable I created a custom view DateRangeSelector which may be used now anywhere one needs to make the user a date-range (currently of course this is only needed in cache filter).

Something like this already exists as material design component: https://material.io/components/date-pickers

> But I haven't done amything real on this up to now anyway, only thing I did so far is creating the new themes dark_material and light_material as copies of dark/light but inheriting from the material-themes instead of the appcompat themes.

I don't know if this is a good idea. I have already created many PR as there were crashes caused by missing attribute definitions in different themes. Why can't we simply switch all appcompat themes to material (which are AFAIK also based on appcompat)?

Reply:
(wrt to work on new base layout)

> So should I remove this stuff for now? (but please in a separate issue)

No, please go ahead! So far I've done a couple of things on resource organization wrt to dark mode / light mode - there is a nice new automatism for it, so that you don't have to apply dark theme / light theme anywhere, but set one config flag, and Android chooses the right resources for you. - But this mechanism does not work with the old preferences implementation we currently use, so that's one of the reasons why we should move forward to using `AppCompatActivity` and `PreferenceFragment` for it.

Another thing I've started working on along the way is placing most of the attribute values for themes/styles into the base theme, which allowed me to cut off a couple of in-between-themes, and also reduces the pain of whether a certain attribute is available in current theme or not.

But all of this can be merged at a later time, no worries.

Reply:
> > In an attempt to make gui components more reusable I created a custom view DateRangeSelector which may be used now anywhere one needs to make the user a date-range (currently of course this is only needed in cache filter).
> 
> Something like this already exists as material design component: https://material.io/components/date-pickers

Saws that and tried it first. It did not work for some reason (renders very strange, selection bubbles are off-place), also when using the "pure" Material Themes. I did not want to spend too muich time just now, and luckily for date/time select we had a rather good picker almost ready (only thing missing was the ability to also "deselect" a value which is neede in context of filter). 

Wrapping a View class around this gives us the ability to later replace the inner works with e.g. the Material-DatePicker without much fiddling elsewhere

Reply:
> > But I haven't done amything real on this up to now anyway, only thing I did so far is creating the new themes dark_material and light_material as copies of dark/light but inheriting from the material-themes instead of the appcompat themes.
> 
> I don't know if this is a good idea. I have already created many PR as there were crashes caused by missing attribute definitions in different themes. Why can't we simply switch all appcompat themes to material (which are AFAIK also based on appcompat)?

This should not happen here any more since the themes are based on Material-Themes, so all necessary attributes are there. Actually this is the way that Material.io wants you to work: make your own theme, but inherit from the Material ones as a base (actually our themes "dark" and "light" are doing the same but inherit from AppCompat - thanks to the work of @moving-bits. Material Themes are also based on AppCombat but have some extensions necessary for Material components as far as I understand).

Reply:
> Another thing I've started working on along the way is placing most of the attribute values for themes/styles into the base theme, which allowed me to cut off a couple of in-between-themes, and also reduces the pain of whether a certain attribute is available in current theme or not.
> 
> But all of this can be merged at a later time, no worries.

Great! Thanks for your work. I did not look into the "auto-dark-mode" setting thing yet.